src/services/directoryServices/MediaDirectoryService.js

```diff
@@ -35,36 +36,32 @@ class MediaDirectoryService {
     if (!isMediaPathValid({ path: directoryName }))
       throw new BadRequestError("Invalid media folder name")
     const mediaType = directoryName.split("/")[0]
-    const { private: isPrivate } = await this.gitHubService.getRepoInfo(
-      sessionData
-    )
-    const files = (
-      await this.listWithDefault(sessionData, { directoryName })
-    ).filter(
-      (file) =>
-        (file.type === "file" || file.type === "dir") &&
-        file.name !== PLACEHOLDER_FILE_NAME
-    )
 
-    const resp = await Promise.all(
-      files.map((curr) => {
-        if (curr.type === "dir") {
-          return {
-            name: curr.name,
-            type: "dir",
-          }
-        }
-        return getMediaFileInfo({
-          file: curr,
-          siteName,
+    if (isGGSWhitelistedRepo(siteName)) {
+      return await this.gitHubService.readMediaDirectory(sessionData, {
+        readFromGithub: false,
```

Comment:
Same concerns here. I think we should consider that non-whitelisted repo == read from GitHub and whitelisted repo == don't read from GitHub.

Reply:
I considered that, but this means we need to have BaseDirectoryService within RepoService as the Github flow is dependent on BaseDirectoryService's `list` method

But I guess it makes sense to keep the whitelisted flow internal to RepoService as well

Reply:
Hmm I think we should be able to cut away the dependency on BaseDirectoryService and utilise the `readDirectory` method in RepoService directly. The return types are similar.